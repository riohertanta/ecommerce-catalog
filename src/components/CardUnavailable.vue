<!-- template tag used for create reusable components contain a container for all html tag -->
<template>
  <!-- div tag for parent unavailable card product section -->
  <div class="container-unavailable">
    <!-- child unavailable card product section -->
    <div class="main-unavailable">
      <!-- p tag used for render description unvailable product except men or women category product -->
      <p class="desc-unavailable">This product is unavailable to show</p>
      <!-- button tag used for even handling with v-on directive attribute while click, renders each card products -->
      <button class="button-unavailable" @click="increment">
        Next product
      </button>
      <!-- this tag used to parent image background unavailable product section -->
      <div class="image-background">
        <img class="face-sad" src="../assets/Sad-Face.svg" :alt="altText" />
      </div>
    </div>
  </div>
</template>

<!-- script tag for interactive ecommerce web vue project -->
<script>
export default {
  // name component for export file
  name: "CardUnavailable",
  data() {
    return {
      altText: "Sad Face image",
    };
  },
  // props declaration for communicates between parent and child components
  props: ["increment"],
};
</script>

<!-- style tag for style web and attribute scoped used for apply to elements of the current component only -->
<style scoped>
/* import file from name folder style */
@import "../assets/style/CardUnavailable.css";
</style>
