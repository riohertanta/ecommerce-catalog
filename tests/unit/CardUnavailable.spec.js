import { shallowMount } from "@vue/test-utils";
import CardUnavailable from "@/components/CardUnavailable.vue";

// describe used to create a block that groups together several related test scenarios
describe("CardUnavailable.vue", () => {
  // test used for test scenario to be tested
  test("should have 'CardUnavailable' as name", () => {
    // assert
    expect(CardUnavailable.name).toBe("CardUnavailable");
  });
  test("should render correctly", () => {
    // arrange
    const wrapper = shallowMount(CardUnavailable);

    // assert
    expect(wrapper.element).toMatchSnapshot();
  });
});
