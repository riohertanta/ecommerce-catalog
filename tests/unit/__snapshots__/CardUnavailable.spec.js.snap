// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardUnavailable.vue should render correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- div tag for parent unavailable card product section -->
  <div
    class="container-unavailable"
  >
    <!-- child unavailable card product section -->
    <div
      class="main-unavailable"
    >
      <!-- p tag used for render description unvailable product except men or women category product -->
      <p
        class="desc-unavailable"
      >
        This product is unavailable to show
      </p>
      <!-- button tag used for even handling with v-on directive attribute while click, renders each card products -->
      <button
        class="button-unavailable"
      >
         Next product 
      </button>
      <!-- this tag used to parent image background unavailable product section -->
      <div
        class="image-background"
      >
        <img
          alt="Sad Face image"
          class="face-sad"
          src=""
        />
      </div>
    </div>
  </div>
  
</div>
`;
